### ISSUE TITLE
채권 투자 리스크 조사

### ISSUE THREAD

**Issue by sangdo14** [sangdo14/bondview_public#2]
- 추천 종목 리포트에 리스크 분석 내용을 추가
- AI 리스크 분석 (머신러닝>추론)

# 채권 투자 리스크 조사
1. 채권 투자 리스크 분석 종류 
2. '신용위험 분석' 과 '중도상환위험' 분석 사례
3. AI가 작성한 데모 코드
4. 결론: 재무재표를 기반으로 신용도를 추론(변경을 예상)하는 AI 개발

_퍼플렉시티, 라이너 이용해서 조사 (2h)_

## 1. 채권 투자 리스크 분석 종류

1. 신용 위험 (Credit Risk): 
발행자의 신용도 변동과 디폴트 가능성, 신용등급 변화 추이, 시장 신용 스프레드(국채 대비 민간 채권 금리 차이) 분석 등을 통해 발행자의 채무불이행 가능성을 평가합니다. 신용평가기관(무디스, S&P, 피치)의 최신 평가와 리포트가 활용됩니다.

2. 중도상환 위험 (Call Risk): 
콜옵션 채권의 경우 발행자의 조기 상환 가능성, 시장금리 변동에 따른 콜 옵션 행사가능성, 콜 리스크를 반영한 기대수익률 분석 및 재투자 위험 설명이 포함됩니다.

3. 유동성 위험 (Liquidity Risk): 
채권 시장에서 매도 시 가격 변동성, 거래량 현황, 시장 깊이 등을 분석하며, 유동성 부족에 따른 가격 하락 가능성을 평가하고 투자자의 매도 전략에 영향을 미치는 요소를 다룹니다.

4. 이자율 위험 (Interest Rate Risk): 
금리 변동 시 채권 가격 변동성(듀레이션, 컨벡서티 등)을 수치화하여 투자 포트폴리오 감도 분석, 금리 상승 시 손실 가능성과 헤지전략을 함께 제시합니다.

5. 구매력 위험 / 인플레이션 위험 (Inflation Risk): 
실질 수익률(명목 수익률에서 인플레이션율 차감) 분석, 인플레이션 연동채권(TIPS 등) 비교, 인플레이션 상승 시 채권 실질가치 저하 가능성을 설명합니다.

6. 재투자 위험 (Reinvestment Risk): 
투자 현금흐름(이자 및 만기 상환금) 재투자 시 금리 변동 가능성과 그에 따른 수익률 변동 분석, 재투자금리 하락 시 손실 시나리오 등을 다룹니다.

7. 환율 위험 (Foreign Exchange Risk): 
해외채권 또는 외화표시 채권 투자 시 환율 변동성, 환헤지 비용, 환율 변동에 따른 현금흐름 변동 위험, 환헤지 전략 제시가 포함됩니다.

## 2. '신용위험 분석' 과 '중도상환위험' 분석 사례
- 위 2가지 리스크 분석이 이해하기 쉽고, 실제로 본인이 채권 투자를 할 때 참고할 수 있을 내용이라고 생각.

**1. 신용위험(Credit Risk) 분석의 AI 적용 사례**

비정형 데이터 활용 평가
- 소셜미디어+재무제표 융합분석: 기업 실적발표회의록 텍스트(KoBERT 감성분석)와 재무비율(유동비율, 부채비율)을 결합해 디폴트 예측. 200개 이상 기업 데이터로 AUC 0.85 달성 사례
- 행동 데이터 기반 평가: 모바일 앱 사용 빈도(초당 583건 로그)를 LSTM으로 분석, 기존 모델 대비 15% 높은 예측 정확도

AI/모델링 관련 시사점
- 신용위험 모델링은 구조적 모델(기업가치 기반)과 축소형 모델(시장 데이터 기반)로 구분되며, 데이터 부족 등 한계가 존재합니다
- 최근에는 신용위험 가격 책정과 포트폴리오 다각화에 있어 고빈도 위험에는 강점을 보이나, 저빈도·고충격 이벤트(금융위기 등)에는 한계가 있음을 지적합니다

XGBoost 기반 디폴트 예측
from xgboost import XGBClassifier
model = XGBClassifier(n_estimators=100, max_depth=3)
model.fit(X_train, y_train)

**2. 중도상환위험(Call Risk) 분석의 AI 기법**

금리 시나리오 예측
- 몬테카를로 시뮬레이션: FRED 미국 국채 수익률 데이터로 1,000개 금리 경로 생성. 콜행사 확률을 P=1N∑i=1NI(ri<rstrike)*P*=*N*1∑*i*=1*NI*(*ri*<*rstrike*) 공식으로 계산
- 트위터 감정지수 연동: 금리 결정 30일 전 소셜미디어 텍스트 분석(r=0.78 상관관계)

AI/모델링 관련 시사점
- 제공된 논문 초록에서는 중도상환 위험(Call Risk)에 대한 직접적 분석이나 AI 적용 사례는 확인되지 않습니다.
- 다만, 신용위험 평가에서 사용되는 다양한 모델(구조적/축소형)과 신용등급, 시장 데이터 분석 기법이 콜옵션 행사 가능성 예측 등 중도상환 위험 분석에도 응용될 수 있음을 시사합니다

import numpy as np
current_rate = 3.5 -- 현재금리
sim_rates = np.random.normal(loc=current_rate, scale=0.2, size=1000)
call_prob = np.sum(sim_rates < 3.0)/1000 -- 행사금리 3% 가정

## 3. AI가 작성한 데모 코드
- 퍼플렉시티, 라이너, 제미나이, 클로드 를 통해 데모 코드 요청 

**데모1**
- 추천데이터셋: Kaggle Credit Risk, DART 기업재무재표, FRED 경제지표
- 데모코드: 기업 유동비율, 부채비율, 영업익률 > 디폴트(파산) 여부
- XGBoost 기반 디폴트 예측

**데모2**
- 추천데이터셋: Lending Club 대출 데이터셋이나 Kaggle의 Home Credit Default Risk
- 데모코드: 개인 대출신청금액, 연간소득, 이자율, 부채대비소득비율 > 디폴트(파산) 여부
- SMOTE로 불균형 문제를 해결한 후 로지스틱 회귀와 랜덤 포레스트 모델로 신용 디폴트 여부를 예측

**데모3**
- 추천데이터셋: 기업 재무제표, 기업 일반정보, 신용등급, 정성적 정보(뉴스, 이슈), 경제지표(금리, 환율, 주가지수, 경제성장률)
- 데모코드: 기업 총자산, 총부채, 매출액, 당기순이익, 영업활동 현금흐름 > 디폴트(파산) 여부
- 로지스틱 회귀

**데모4**
- 추천데이터셋: German Credit Data, Default of Credit Card Clients, Lending Club Data, South German Credit Dataset
- 데모코드: 개인 신용이력, 고용상태, 당좌예금, 대출금액, 할부비율, 저축상태, 개인속성, 자산보유, 거주형태... > 주요 위험요인 도출
- 로지스틱 회귀, 랜덤포레스트, Gradient Boosting, SVM ( 최신 알고리즘 XGBoost, LightGBM )

## 4. 멘토님과 내린 결론
- AI를 이용해 매매 의사결정 및 의견을 내리는 것은 불확실성이 크다.
- 좀 더 신빙성 있는 데이터를 기반으로 투자 결정보다는 가이드 역할을 해줄 수 있는 방향으로 AI를 활용
- 결론: 재무재표를 기반으로 신용도를 추론(변경을 예상)하는 AI
